test/rules/invariant_booleans.dart

```diff
@@ -237,3 +237,13 @@ void bug372(bool foo) {
     // doSomethingElse();
   }
 }
+
+void bug337(int offset, int length) {
+  if (offset >= length) {
+    return;
+  }
+
+  offset++;
```

Comment:
This is a very good idea, but I am not sure which ones to check, can you please provide a few examples?